qiling/extensions/r2/r2.py

```diff
@@ -12,21 +12,27 @@
 from qiling.core import Qiling
 
 @dataclass(unsafe_hash=True)
-class Function:
+class R2Data:
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
```

Comment:
There is no need to decorate `R2Data` subclasses with `@dataclass` anymore, they all inherit that from `R2Data`.